components/os/os.go

```diff
@@ -19,9 +19,11 @@ const (
 )
 
 type AgentVersion struct {
-	Major       string
-	Minor       string // Empty means latest
-	BetaChannel bool
+	Major         string
+	Minor         string // Empty means latest
+	BetaChannel   bool
+	RepoBranch    string
+	IsCustomImage bool
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
These 2 options are valid only together. They could merge to a unique property `PipelineId`, and parser would go like

 ```go
version, err := parseVersion("pipeline-16362517-a7")
assert.NoError(t, err)
assert.Equal(t, version, os.AgentVersion{
  PipelineId:    "16362517",
  Major: "7",
  Minor: "",
  BetaChannel: false
})
```

Reply:
Not related to this PR, I wonder why `AgentVersion` is here, it should belong to agent packages

Reply:
agreed, removed the redundant variable and changed the name as you suggested